app/controllers/units_controller.rb

```diff
@@ -0,0 +1,35 @@
+class UnitsController < ApplicationController
+  def create
+    @system = System.find(params[:unit][:system_id])
+    @unit = Unit.new(unit_params)
+    @unit.destination = @system
+
+    if @unit.save
+      flash[:notice] = "Unit Created"
+    else
+      flash[:notice] = "Failed to Create Unit"
+    end
+
+    respond_to do |format|
+      format.html { redirect_to system_path(@system) }
```

Comment:
what happens if I run a browser without JavaScript (screen readers, etc). Good practice would be to handle for a gracefully degraded experience (one without ajax)